app/lib/actions/pre_upgrade_action.rb

```diff
@@ -0,0 +1,41 @@
+module Actions
+  module ForemanLeapp
+    class PreUpgradeAction < Actions::EntryAction
+      def self.subscribe
+        Actions::RemoteExecution::RunHostJob
+      end
+
+      def plan(job_invocation, host, template_invocation, *args)
+        return unless job_invocation.job_category == ::ForemanLeapp::JOB_CATEGORY
+        binding.pry
+        plan_self(trigger_success: triggering_action.output[:success],
+                  run_host_job_task_id: template_invocation.run_host_job_task_id,
+                  host_id: host.id)
+      end
+
+      def run
+        self.output.update(success: true) if self.input[:trigger_success]
+      end
+
+      def finalize(*args)
+        host = Host.find(self.input[:host_id])
+        task = ForemanTasks::Task.find(self.input[:run_host_job_task_id])
+        leapp_report = format_output(task.main_action.continuous_output.humanize)
+
+        PreupgradeReport.create_report(host, leapp_report)
+      end
+
+      private
+
+      def format_output(job_output)
+        lines = job_output.split(/\n/)
```

Comment:
Why split using a regular expression describing a new line when we have things like `String#lines` and `String#each_line`?